core/tests/protractor_utils/ExplorationEditorTranslationTab.js

```diff
@@ -31,12 +31,21 @@ var ExplorationEditorTranslationTab = function() {
   this.exitTutorial = function() {
     // If the translation welcome modal shows up, exit it.
     translationWelcomeModal.isPresent().then(function(isVisible) {
-      if (isVisible) {
-        waitFor.elementToBeClickable(
-          dismissWelcomeModalButton,
-          'Welcome modal is taking too long to appear');
-        dismissWelcomeModalButton.click();
-      }
+      waitFor.visibilityOf(dismissWelcomeModalButton, 2000,
+        'Welcome modal not becoming visible').then(
+        () => {
+          waitFor.elementToBeClickable(
+            dismissWelcomeModalButton,
+            'Welcome modal is taking too long to appear');
+          dismissWelcomeModalButton.click();
+        },
+        (err) => {
+          // This is just an empty error function to catch when the
+          // the welcome modal has been dismissed once. If this is not present
+          // then protractor uses the default error function which is not
+          // as this is not an error.
```

Comment:
I like this solution!

Could you make this comment a bit clearer? In particular:
* Clarify that when the modal has been dismissed once before, it won't appear, so the `waitFor` will time out
* I think you're missing a word at the end of line 45, maybe "correct"?

Reply:
Done!